src/main/java/com/example/nameservice/NameController.java

```diff
@@ -0,0 +1,27 @@
+package com.example.nameservice;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+public class NameController {
+
+    private NameMapper nameMapper;
+
+    public NameController(NameMapper nameMapper) {
+        this.nameMapper = nameMapper;
+    }
+
+    @GetMapping("/names")
+
+    public List<Name> findNames(@RequestParam(required = false) String startsWith) {
+        if (startsWith != null) {
```

Comment:
という書き方もありますので参考までに...
"non null"とより自然言語に近いので可読性は高いです。
```suggestion
        if (Objects.nonNull(startsWith)) {
```